manager/constraint/constraint.go

```diff
@@ -0,0 +1,164 @@
+package constraint
```

Comment:
Yeah, it's a moot point. To me it felt like it was part of the scheduler (which at the end of the day is the component responsible to place tasks onto nodes), and it turns out another component (global orchestrator) wants to use a piece of the scheduler. I don't think the scheduler and orchestrator are sharing a common piece, more like the orchestrator wants to use the scheduler for some stuff.

But it's a minor personal preference, no objective arguments